armory/paths.py

```diff
@@ -9,58 +9,132 @@
 logger = logging.getLogger(__name__)
 
 
-def _parse_global_config(path, armory_dir):
-    if os.path.exists(path):
-        try:
-            with open(path) as f:
-                config = json.load(f)
-        except json.decoder.JSONDecodeError:
-            logger.exception(f"Armory config file {path} could not be decoded")
-            raise
-        except OSError:
-            logger.exception(f"Armory config file {path} could not be read")
-            raise
-    else:
-        os.makedirs(armory_dir, exist_ok=True)
-        config = {
-            "cached_dataset_dir": os.path.join(armory_dir, "datasets"),
-            "output_dir": os.path.join(armory_dir, "outputs"),
-        }
-        with open(path, "w") as f:
-            json.dump(config, f)
+# Only initialize the host and docker paths once
+_MAP = {}
+
+
+def default():
+    if "default" not in _MAP:
+        _MAP["default"] = HostDefault()
+    return _MAP["default"]
+
+
+def host():
+    if "host" not in _MAP:
+        _MAP["host"] = HostPaths()
+    return _MAP["host"]
+
+
+def docker():
+    if "docker" not in _MAP:
+        _MAP["docker"] = DockerPaths()
+    return _MAP["docker"]
+
+
+def init():
+    """
+    Load configurations
+    """
+    default()
+    host()
+    docker()
```

Comment:
Could you describe why an init method is needed? Can't we just call default / host / docker as needed?

Reply:
Yeah, that makes sense. The only place I use it is in `__main__`, but with the `paths.host().makedirs()` lines in main, it's redundant. Removed.